### DISCUSSION TITLE
Any way for optional elements in a query at runtime ?

### DISCUSSION THREAD

**Discussion by rfriend-zilo** [byme8/ZeroQL#68] [General]
I have tried doing this inline something like this, i have combinations of these types of things and am just providing an example, but at runtime how can i include/exclude parts/fragments based on parameters ?

`roles = includeRoles ?  user.Roles(null, r => r.AsRoleFragment()) : null,`

and also as a fragment

```
    [GraphQLFragment]
    public static UserModelFragment AsUserFragment(this UserModel userModel, bool includeRoles)
    {
        return new UserModelFragment
        {
            Roles = includeRoles : userModel.Roles(r => r.AsRoleFragment()) : null,
            etc..
        };
    }

```

**Comment by byme8** [comment]
Fragments are analyzed at compile time to achieve best performance possible. There is no way to change it at runtime.

**Comment by rfriend-zilo** [comment]
Can this be done when not using fragments ?

Currently my client library / sdk looks like this

`await Service.GetSomethingsWithRelationship1()`
`await Service.GetSomethingsWithRelationship2()`
`await Service.GetSomethingsLight()`

What i would dearly love is something like this using an enum with `[Flags]` attribute

`await Service.GetSomethings(SomethingsRelationships.Relationship1 | SomethingsRelationships.Relationship2)`


And construct the GQL query accordingly depending on the flags passed in.

Im using HotChocolate resolvers to then load related data when requested.

Is there anyway that this could be done at all ?
